tests/validators/test_allow_partial.py

```diff
@@ -0,0 +1,272 @@
+            },
+            total=False,
+        )
+    )
+
+    assert v.validate_python({'a': 11, 'b': '12', 'c': 13}) == snapshot({'a': 11, 'b': 12, 'c': 13})
+    with pytest.raises(ValidationError, match='Input should be greater than 10'):
+        v.validate_python({'a': 11, 'b': '12', 'c': 1})
+    assert v.validate_python({'a': 11, 'b': '12', 'c': 1}, allow_partial=True) == snapshot({'a': 11, 'b': 12})
+    with pytest.raises(ValidationError, match=r'b\s+Field required'):
+        v.validate_python({'a': 11, 'c': 12}, allow_partial=True)
+    with pytest.raises(ValidationError, match=r'b\s+Input should be greater than 10'):
+        v.validate_python({'a': 11, 'c': 12, 'b': 1}, allow_partial=True)
+
+
+def test_double_nested():
+    v = SchemaValidator(
+        core_schema.typed_dict_schema(
+            {
+                'a': core_schema.typed_dict_field(core_schema.int_schema(gt=10)),
+                'b': core_schema.typed_dict_field(
+                    core_schema.list_schema(
+                        core_schema.dict_schema(core_schema.str_schema(), core_schema.int_schema(ge=10))
+                    )
+                ),
+            },
+            total=False,
+        )
+    )
+    assert v.validate_python({'a': 11, 'b': [{'a': 10, 'b': 20}, {'a': 30, 'b': 40}]}) == snapshot(
+        {'a': 11, 'b': [{'a': 10, 'b': 20}, {'a': 30, 'b': 40}]}
+    )
+    assert v.validate_python({'a': 11, 'b': [{'a': 10, 'b': 20}, {'a': 30, 'b': 4}]}, allow_partial=True) == snapshot(
+        {'a': 11, 'b': [{'a': 10, 'b': 20}, {'a': 30}]}
+    )
+    assert v.validate_python({'a': 11, 'b': [{'a': 10, 'b': 20}, {'a': 30, 123: 4}]}, allow_partial=True) == snapshot(
+        {'a': 11, 'b': [{'a': 10, 'b': 20}]}
+    )
+    # the first element of the list is invalid, so the whole list is invalid
+    assert v.validate_python({'a': 11, 'b': [{'a': 10, 'b': 2}, {'a': 30}]}, allow_partial=True) == snapshot({'a': 11})
+    with pytest.raises(ValidationError, match=r'b\.0\.b\s+Input should be greater than or equal to 10'):
+        v.validate_python({'b': [{'a': 10, 'b': 2}, {'a': 30}], 'a': 11}, allow_partial=True)
+
+    with pytest.raises(ValidationError, match=r'b\.1\.a\s+Input should be greater than or equal to 10'):
+        v.validate_python({'b': [{'a': 10, 'b': 20}, {'a': 3}], 'a': 11}, allow_partial=True)
+
+    assert v.validate_python({'a': 11, 'b': [{'a': 1, 'b': 20}, {'a': 3, 'b': 40}]}, allow_partial=True) == snapshot(
+        {'a': 11}
+    )
+    json = b'{"a": 11, "b": [{"a": 10, "b": 20}, {"a": 30, "b": 40}]}'
+    assert v.validate_json(json, allow_partial=True) == snapshot(
+        {'a': 11, 'b': [{'a': 10, 'b': 20}, {'a': 30, 'b': 40}]}
+    )
+    for i in range(1, len(json)):
+        value = v.validate_json(json[:i], allow_partial=True)
+        assert isinstance(value, dict)
+
+
+def test_tuple_list():
+    """Tuples don't support partial, so behaviour should be disabled."""
```

Comment:
Why not? At least for variadic tuples this seems potentially acceptable.

Reply:
I definitely want to support it in future, I was just implementing it for the minimum set of validators to prove the idea initially.